### ISSUE TITLE
Problem with vcvars_command and x86_64 only build

### ISSUE THREAD

**Issue by oakaf** [conan-io/conan#1253]
I have a problem with using the vcvars_command in for a x86_64 restricted build recipe (it's a recipe for OpenSSL, but I guess this is general behaviour).
The restriction is done in recipe settings -> "arch": {"x86_64"}, as we only need the 64-bit versions.
Withing the "build"-step I then use 
```
env_build = VisualStudioBuildEnvironment(self)
with tools.environment_append(env_build.vars):
    vcvars = tools.vcvars_command(self.settings)
```

but here the vsvars_command gives me an error 
`Invalid setting 'x86' is not a valid 'settings.arch' value. Possible values are ['x86_64']`.

The trace refers back to line 165 in conans/tools.py and line 73 in conans/model/settings.py (Conan 0.22.2, but also occured in 0.20.2; tested also on two different machines).

If I include "x86" in the arch settings, this does not happen, but this is also not what I want to do.

Is this an intended behaviour or is there any solution for this?

**Comment by memsharded** [comment]
Hi @oakaf ,

Just like a quick tip, I recently prefer not restricting the recipe settings, but instead launching an Exception for non supported platforms, like:

```def configure(self):
      if self.settings.arch != "x86_64":
             raise Exception("Openssl only supported for 64 bit versions")
```

Besides that tip, yeah, seems like a bug, the vcvars_command should raise for that, but safely ignore it. Lets try to fix it for next release.

**Comment by oakaf** [comment]
Hm, I thought this is (more or less) what Conan already internally does, when you set some restrictions? At least I get such warnings, if I restrict the recipe to an OS or compiler version.

Shouldn't it be the goal, to put as least logic as possible into the recipe and let Conan handle as most as possible? Or are there any (bad) implications of these restrictions, which I miss at the moment?

Putting all those restrictions in the recipe settings would also have the advantage, that you can directly see, for what specific system the recipe has been written, instead of going through the (build-) steps and looking for all those if-statements.

Ok, but apart from that, thanks for the tip, I will include it in my recipes.

**Comment by memsharded** [comment]
Yes, you are totally right. I said it was my personal preference, so it is OK that you are using the constraints, to make it more evident, it was a conan failure. This is why I annotated it as a bug, and planned to address it for next release :) Thanks for the feedback!

**Comment by memsharded** [comment]
Released in 0.23, please upgrade and try it. Thanks!
